### DISCUSSION TITLE
comparison to ramda?

### DISCUSSION THREAD

**Discussion by xddq** [gcanti/fp-ts#1580] [Q&A] [answered]
I am trying to get my head around functional programming and I don't know which library I should use (or start with).

Do you happen to know [ramda](https://github.com/Ramda/ramda) and could draw a line? When to use which library? Which one for starting, which one for advanced fp, and why?

thank you in advance!

**Comment by samhh** [comment, marked as answer]
Ramda isn't very well typed in TypeScript in some places. fp-ts takes a bit more inspiration from Haskell. You can find an API reference/comparison [here](https://samhh.github.io/fp-ts-std/ramda).

**Reply by xddq** [reply]
Interesting, thanks for the link and the quick response. I think I'll dive into fp-ts.
